ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/DiscretePstTapFiller.java

```diff
@@ -48,9 +48,9 @@ public DiscretePstTapFiller(Network network,
 
     @Override
     public void fill(LinearProblem linearProblem, FlowResult flowResult, SensitivityResult sensitivityResult) {
-        rangeActions.forEach((state, rangeActionSet) -> rangeActionSet.forEach(rangeAction ->
-            buildPstTapVariablesAndConstraints(linearProblem, rangeAction, state)
-        ));
+        rangeActions.entrySet().stream()
+            .sorted(Comparator.comparingInt(e -> e.getKey().getInstant().getOrder())).forEach(entry -> entry.getValue().forEach(rangeAction ->
```

Comment:
Because to construct my constraint I need preventive state information when I'm in curative state. It's the same thing that was done inside the `CoreProblemFiller` method : `buildRangeActionConstraints`